packages/data/src/types/driver.ts

```diff
@@ -0,0 +1,8 @@
+import type { AbstractQuery } from './abstract-query.js';
+
+export abstract class DataDriver {
+	abstract connect: () => Promise<void>;
```

Comment:
due to [this comment](https://github.com/directus/directus/pull/18642#discussion_r1218355265) it seems that the `connect` function  doesn't need to be within each driver.

and similarly with the discussion about having a `client` in the abstract driver, we also might not need the `disconnect` function here, e.g. for a driver which talks to the datastore via stateless HTTP requests.

Reply:
Even though it's not needed, I wanna keep it as a required (defaults to no-op) method, so we can simply do a `registeredDrivers.forEach(driver => driver.disconnect())` in the main data manager. That being said, the names `connect` / `disconnect` might not be the correct naming. I'd imagine that a http-adapter could use the `.connect()` as the point where it authenticates the configured user or something. Lemme noodle on the name a bit